### PR TITLE
Add implementation of the command finder

### PR DESCRIPTION
**Pull request by neithernut**
TThis PR implements `wl_command_get()`.

### CODE REVIEW

File: src/command/command.c [hunk 1]
```diff
@@ -34,11 +39,47 @@
...
 ) {
-    //!< @todo: implement
+    // initialize bounds
+    size_t first = 0;
+    size_t alast = ws_command_cnt;
+    size_t cur;
+
+    // perform a binary search
+    while (first + LINEAR_THRESHOLD > alast) {
+        cur = first+alast / 2;
```
Comment (matthiasbeyer) [inline src/command/command.c]: Please add spaces around `+`.

File: src/command/command.c [hunk 2]
```diff
@@ -34,11 +39,47 @@
...
+
+        // compare the current name with what we want
+        int cmp = strcmp(name, ws_command_list[cur].name);
+
+        // the name we want is smaller
+        if (cmp<0) {
```
Comment (matthiasbeyer) [inline src/command/command.c]: Please add spaces around `<`.

File: src/command/command.c [hunk 3]
```diff
@@ -34,11 +39,47 @@
...
+            alast = cur;
+            continue;
+        }
+
+        // the name we want is greater
+        if (cmp>0) {
```
Comment (matthiasbeyer) [inline src/command/command.c]: Please add spaces around `>`.

File: src/command/command.c [hunk 4]
```diff
@@ -34,11 +39,47 @@
...
+
+        // bullseye
+        return ws_command_list+cur;
+    }
+
+    // we are now in a realm where linear search is expected to be faster
+    cur = alast;
+    while (cur-- > first) {
+        if (strcmp(name, ws_command_list[cur].name) == 0) {
+            return ws_command_list+cur;
```
Comment (matthiasbeyer) [inline src/command/command.c]: Please add spaces around `+`.

File: src/command/command.c [hunk 5]
```diff
@@ -34,11 +39,47 @@
...
+        }
+
+        // the name we want is greater
+        if (cmp > 0) {
+            first = cur + 1;
+            continue;
+        }
+
+        // bullseye
+        return ws_command_list+cur;
```
Comment (matthiasbeyer) [inline src/command/command.c]: Missing spaces around `+`.

**Comment by neithernut** [comment]
Look again.

**Comment by matthiasbeyer** [comment]
ACK.

**Pull request merged**